DfE.FindInformationAcademiesTrusts/Pages/Cookies.cshtml.cs

```diff
@@ -0,0 +1,117 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.IdentityModel.Tokens;
+
+namespace DfE.FindInformationAcademiesTrusts.Pages;
+
+public class CookiesModel : PageModel
+{
+    public static string ConsentCookieName = ".FindInformationAcademiesTrust.CookieConsent";
+    public static string DeleteCookieTempDataName = "DeleteCookie";
+    public static string CookieChangedTempDataName = "CookieResponse";
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    public bool DisplayCookieChangedMessageOnCookiesPage { get; set; }
+    [BindProperty(SupportsGet = true)] public string? ReturnPath { get; set; }
+    [BindProperty(SupportsGet = true)] public bool? Consent { get; set; }
+
+    public CookiesModel(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    /// <summary>
+    /// Used to load the cookies page or to set cookie preferences from the cookie banner
+    /// </summary>
+    /// <returns>The cookies page if we are not setting the cookie preference, otherwise redirect to the redirect path (from the cookie banner)</returns>
+    public ActionResult OnGet()
+    {
+        ValidateReturnPath();
+
+        ApplyCookieConsent();
+
+        if (ReturnPath.IsNullOrEmpty() || Consent is null)
+        {
+            if (CookiesPreferencesHaveBeenSet())
+            {
+                Consent = OptionalCookiesAreAccepted(_httpContextAccessor.HttpContext!, TempData);
+            }
+
+            return Page();
+        }
+
+        return Redirect(ReturnPath!);
```

Comment:
Use `LocalRedirect` here to be super-sure that we're not redirecting to an attackers website.
https://learn.microsoft.com/en-us/aspnet/core/security/preventing-open-redirects?view=aspnetcore-8.0